R/error_on_missing_locations.R

```diff
@@ -1,7 +1,19 @@
+#' Error on missing locations in hierarchy
+#'
+#' Compares data to location hierarchy to ensure that data has values present for each hierarchy entry.
+#' Issues a \code{stop} if any values are missing.
+#'
+#' @param data vector-like with values corresponding to the location hierarchy
+#' @param ... captures other arguments. Do not provide any or you will get an error.
```

Comment:
Will R just let you shove extra args in if you don't have this?

Reply:
No. You can provide too few arguments and potentially surprise yourself, but extra arguments require `...`

```
# a function. takes one arg. what could be simpler?
> one.arg <- function(a) { return(4) }
# see, you call it and you get your result
> one.arg(1)
[1] 4
# oh, the argument is ... optional?
# detail: it only errors when you try and use it. arguments that are never used won't error!
> one.arg()
[1] 4
# name args - ooks normal
> one.arg(a = a)
[1] 4
# we don't take b's in this here function
> one.arg(b = a)
Error in one.arg(b = a) : unused argument (b = a)
# nor do we take additional arguments
> one.arg(1, 4)
Error in one.arg(1, 4) : unused argument (4)
```

somewhat related - argument unpacking is a bit smarter than it is in Python

```
> polarize <- function(a, b, c) {
+ return(list(a=a, b=b, c=c))
+ }
# you might think you'd get an error for assigning two values to a. you'd be wrong
> polarize(2, 3, a = 1)
$a
[1] 1

$b
[1] 2

$c
[1] 3
# i expect you either love or hate this behavior
> polarize(a = 1, 3, b = 2)
$a
[1] 1

$b
[1] 2

$c
[1] 3
```

Reply:
"Smarter"